### PR TITLE
[IMP] website_sale: avoid useless hashes in product urls

### PR DESCRIPTION
**Pull request by Feyensv**
---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_sale/models/product_product.py [hunk 1]
```diff
@@ -69,8 +69,11 @@ def _compute_base_unit_name(self):
     @api.depends('product_tmpl_id.website_url', 'product_template_attribute_value_ids')
     def _compute_product_website_url(self):
         for product in self:
-            attributes = ','.join(str(x) for x in product.product_template_attribute_value_ids.product_attribute_value_id.ids)
-            product.website_url = "%s#attribute_values=%s" % (product.product_tmpl_id.website_url, attributes)
+            pavs = product.product_template_attribute_value_ids.product_attribute_value_id
+            url = product.product_tmpl_id.website_url
+            if pavs:
+                url = "%s#attribute_values=%s" % (url, ','.join(str(x.id) for x in pavs))
```
Comment (AntoineVDV) [inline addons/website_sale/models/product_product.py:75]: Clearer?
```suggestion
            url = product.product_tmpl_id.website_url
            if pavs := product.product_template_attribute_value_ids.product_attribute_value_id:
                pav_ids = [str(pav.id) for pav in pavs]
                url = f'{url}#attribute_values={",".join(pav_ids)}'
```

File: addons/website_sale/static/src/js/website_sale.js [hunk 2]
```diff
@@ -171,7 +171,9 @@ export const WebsiteSale = publicWidget.Widget.extend(VariantMixin, cartHandlerM
         );
         let attributeIds = [];
         inputs.forEach((element) => attributeIds.push(element.dataset.attributeValueId));
-        window.location.hash = 'attribute_values=' + attributeIds.join(',');
+        if (attributeIds.length > 0) {
+            window.location.hash = 'attribute_values=' + attributeIds.join(',');
```
Comment (AntoineVDV) [inline addons/website_sale/static/src/js/website_sale.js:175]:
```suggestion
            window.location.hash = `attribute_values=${attributeIds.join(',')}`;
```

**Review by AntoineVDV — APPROVED** [review]

File: addons/website_sale/models/product_product.py (hunk 1 above)
Comment (Feyensv) [reply addons/website_sale/models/product_product.py]: Mhm, I don't find it clearer, but not really worse either, so no strong opinion, suggestion applied.

**Review by livaios — APPROVED** [review]

**Comment by Feyensv** [comment]
@robodoo r+

**Pull request closed without merging**